pkgpanda/build/src_fetchers.py

```diff
@@ -220,39 +220,49 @@ def _identify_archive_type(filename):
     return 'unknown'
 
 
-def _check_components_sanity(path):
+def _check_components_sanity(path, allow_extra_files):
     """Check if archive is sane
 
-    Check if there is only one top level component (directory) in the extracted
-    archive's directory.
+    Check if there is only one top level component in the extracted
+    archive, and make sure that component is a directory. If
+    'allow_extra_files' is passed, then we are allowed to have other files at
+    the top level (but we still must have only 1 directory).
 
     Args:
         path: path to the extracted archive's directory
+        allow_extra_files: if True, allow other files next to the top level directory
 
     Raises:
         Raise an exception if there is anything else than a single directory
     """
-    dir_contents = os.listdir(path)
+    files = os.listdir(path)
 
-    if len(dir_contents) != 1 or not os.path.isdir(os.path.join(path, dir_contents[0])):
+    if len(files) == 0:
+        raise ValidationError("Extracted archive is empty")
+
+    dirs = [f for f in files if os.path.isdir(os.path.join(path, f))]
+
+    if len(dirs) > 1 or (len(files) > 1 and not allow_extra_files):
```

Comment:
The original code handled 0 directories by raising this `ValidationError`. The new code returns successfully if the archive contains 0 directories. Change `len(dirs) > 1` to `len(dirs) != 1`.

Reply:
This stanza above should already cover that case:

```python
    if len(files) == 0:
        raise ValidationError("Extracted archive is empty")
```

Reply:
Consider the case that `allow_extra_files` is `True` and `path` contains one or more (non-directory) files.